### PR TITLE
ignoreHotspottingChecks if hashFirstRowComponent

### LINKED ISSUE #979: Should not have to ignoreHotspottingWarnings if hashFirstRowComponent is set
This stops hotspotting from happening, right?

```
    /**
     * Prefix the row with a hash of the first row component
     * <p>
     * This helps to ensure that rows are evenly distributed. In particular, using strings as the first row component
     * will only cover the entire range of byte arrays because they're encode with UTF_8. In addition, using any
     * variable-length {@link ValueType} as the first row component type will not work because they are prefixed by the
     * length of the component. Finally, we can't use BLOB if there are multiple components because it must go at the
     * end of the row
     */
```

### PR DESCRIPTION
**Pull request by hsaraogi**
Fixes #979

---
This change is [Reviewable](https://reviewable.io/reviews/palantir/atlasdb/1091)

### CODE REVIEW

File: atlasdb-client/src/test/java/com/palantir/atlasdb/table/description/SchemaHotspottingTest.java [hunk 1]
```diff
@@ -34,46 +34,122 @@
...
+        return tableFirstRowComponentHashed;
+    }
+
+    private static Schema getIndexHotspottingSchema() {
+        Schema suffersFromIndexHotspoting = getIgnoredHotspottingSchema();
+        suffersFromIndexHotspoting.addIndexDefinition("hotSpottingIndex", new IndexDefinition(IndexDefinition.IndexType.CELL_REFERENCING) {{
+            onTable(IGNORED_HOTSPOTTING_TABLE);
+            rowName();
+                componentFromRow(HOTSPOTTER_ROW_COMPONENT, ValueType.VAR_STRING);
+            rangeScanAllowed();
```
Comment (jboreiko) [inline atlasdb-client/src/test/java/com/palantir/atlasdb/table/description/SchemaHotspottingTest.java]: Why is this included out of interest?

File: atlasdb-client/src/test/java/com/palantir/atlasdb/table/description/SchemaHotspottingTest.java [hunk 2]
```diff
@@ -34,46 +34,122 @@
...
+    }
+
+    private static Schema getIgnoredIndexHotspottingSchema() {
+        Schema ignoredIndexHotspottingSchema = getIgnoredHotspottingSchema();
+        ignoredIndexHotspottingSchema.addIndexDefinition("ignoreHotSpottingIndex", new IndexDefinition(IndexDefinition.IndexType.CELL_REFERENCING) {{
+            ignoreHotspottingChecks();
+            onTable(IGNORED_HOTSPOTTING_TABLE);
+            rowName();
+                componentFromRow(HOTSPOTTER_ROW_COMPONENT, ValueType.VAR_STRING);
+            rangeScanAllowed();
```
Comment (jboreiko) [inline atlasdb-client/src/test/java/com/palantir/atlasdb/table/description/SchemaHotspottingTest.java]: Ditto

File: atlasdb-client/src/test/java/com/palantir/atlasdb/table/description/SchemaHotspottingTest.java [hunk 3]
```diff
@@ -34,46 +34,122 @@
...
+    }
+
+    private static Schema getIndexFirstRowComponentHashedSchema() {
+        Schema indexFirstRowComponentHashed = getIgnoredHotspottingSchema();
+        indexFirstRowComponentHashed.addIndexDefinition("firstRowComponentHashedIndex", new IndexDefinition(IndexDefinition.IndexType.CELL_REFERENCING) {{
+            onTable(IGNORED_HOTSPOTTING_TABLE);
+            rowName();
+                hashFirstRowComponent();
+                componentFromRow(HOTSPOTTER_ROW_COMPONENT, ValueType.VAR_STRING);
+            rangeScanAllowed();
```
Comment (jboreiko) [inline atlasdb-client/src/test/java/com/palantir/atlasdb/table/description/SchemaHotspottingTest.java]: Last one

File: atlasdb-client/src/test/java/com/palantir/atlasdb/table/description/SchemaHotspottingTest.java [hunk 4]
```diff
@@ -34,46 +34,122 @@
...
+    }
+
+    @Test
+    public void testNoFailureWhenTableFirstRowComponentHashed() {
+        getTableFirstRowComponentHashedSchema().validate();
+    }
+
+    @Test
+    public void testNoFailureWhenIndexFirstRowComponentHashed() {
+        getIndexFirstRowComponentHashedSchema().validate();
```
Comment (SerialVelocity) [inline atlasdb-client/src/test/java/com/palantir/atlasdb/table/description/SchemaHotspottingTest.java]: We use both `getIgnoredIndexHotspottingSchema()` and `indexHotspottingSchema`. Is there a reason why?

File: atlasdb-client/src/test/java/com/palantir/atlasdb/table/description/SchemaHotspottingTest.java [hunk 5]
```diff
@@ -34,46 +34,122 @@
     @Rule
     public TemporaryFolder temporaryFolder = new TemporaryFolder();
 
+    private static final String HOTSPOTTING_TABLE = "HotspottingTable";
+    private static final String IGNORED_HOTSPOTTING_TABLE = "IgnoredHotspottingTable";
+    private static final String HOTSPOTTER_ROW_COMPONENT = "Hotspotter";
```
Comment (SerialVelocity) [inline atlasdb-client/src/test/java/com/palantir/atlasdb/table/description/SchemaHotspottingTest.java]: These would probably be better as `TABLE_NAME`, `ROW_COMPONENT_NAME` since the values don't really make any difference.

File: atlasdb-client/src/test/java/com/palantir/atlasdb/table/description/SchemaHotspottingTest.java (hunk 4 above)
Comment (hsaraogi) [inline atlasdb-client/src/test/java/com/palantir/atlasdb/table/description/SchemaHotspottingTest.java]: ``indexHotspottingSchema`` is used multiple times, so it looked good to have the variable. 
``getIgnoredIndexHotspottingSchema()`` returns schema where hotspotting is ignored while ``indexHotspottingSchema`` hasnt ignored the check.
